scripts/us_epa/facility/process_facility.py

```diff
@@ -156,12 +145,67 @@ def _validate_geo(facility, dcid, lat, lng):
     point = geometry.Point(float(lng), float(lat))
     polygon = geometry.shape(json.loads(gj))
     if not polygon.contains(point):
-        print(f'Facility {facility} has ({lat}, {lng}) outside entity {dcid}')
         return False
 
     return True
 
 
+_COUNTERS = {
+    'given_county_wrong_latlng': [],
+    'given_county_correct_latlng': [],
+    'zipbased_county_wrong_latlng': [],
+    'zipbased_county_correct_latlng': [],
+    'missing_zip_and_county': [],
+}
+
+
+def _resolve_places(facility_id, zip, provided_county, lat, lng):
+    """Resolve the geo relations for the given Facility
+
+    Returns resolved <zip>, <county>, <lat>, <lng>
+    """
+    if zip == 'zip/00000':
+        _COUNTERS['missing_zip_and_county'].append(facility_id)
+        return '', '', '', ''
+
+    county_candidates = _get_county_candidates(zip)
+    if any(provided_county in l for l in county_candidates):
+        # Provided county is in the candidate list, use that.
+
+        if lat and lng and _validate_latlng(lat, lng, provided_county):
+            # Lat/lng is in the chosen county
+            _COUNTERS['given_county_correct_latlng'].append(facility_id)
+            return zip, provided_county, lat, lng
+
+        _COUNTERS['given_county_wrong_latlng'].append(facility_id)
```

Comment:
have you tried skipping this, and falling through if there is lat/lng defined (but the wrong county is entered?).

Reply:
This is the case where the county entered is likely right (because DC zip containment matches), but the lat/lng is slightly off.  I thought its better to use the county they report for stats, even if we skip the lat/lng (the alternative is to potentially pick a different zip-overlapping county with right lat/lng), and there are only 66 such cases.

But if the wrong county is entered, then the check in line 172 will not pass, so we will validate the candidate counties against lat/lng in the block below.

Reply:
gotcha - thanks sgtm!